### PR TITLE
[Merged by Bors] - refactor: split `BumpFunctionInner`

### LINKED ISSUE #4755: Split `Analysis.Calculus.BumpFunctionInner`
There are several sections in this file that can be moved to different
files to reduce imports:

- `Analysis.Calculus.SmoothTransition`: definitions and lemmas about
  `Real.expInvGlue` and `Real.smoothTransition`.
- `Analysis.Calculus.BumpFunction.Basic`: definition of the typeclass
  `HasContDiffBump` and properties of `ContDiffBump`.
- `Analysis.Calculus.BumpFunction.Inner`: an instance for inner
  product spaces.
- `Analysis.Calculus.BumpFunction.Normed`: definition and lemmas about
  `BumpFunction.normed`; this is the only file that will import
  measure theory.

### PR DESCRIPTION
**Pull request by urkud**
Fixes #4755 

---
Should not be merged before we add the `port-complete` tag.

@sgouezel @fpvandoorn I tried to get copyright headers right from `git blame`. Feel free to edit them.

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Analysis/Calculus/BumpFunctionInner.lean
Comment (fpvandoorn) [inline Mathlib/Analysis/Calculus/BumpFunctionInner.lean:1]: should this file move to the new subfolder?

File: Mathlib/Analysis/Calculus/BumpFunction/Basic.lean [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+! This file was ported from Lean 3 source module analysis.calculus.bump_function_inner
+! leanprover-community/mathlib commit 3bce8d800a6f2b8f63fe1e588fd76a9ff4adcebe
+! Please do not edit these lines, except to modify the commit id
+! if you have ported upstream changes.
+-/
+import Mathlib.Analysis.Calculus.ContDiff
+import Mathlib.Analysis.NormedSpace.FiniteDimension
+
+/-!
+# Infinitely continuous "bump" functions
```
Comment (fpvandoorn) [inline Mathlib/Analysis/Calculus/BumpFunction/Basic.lean:15]:
```suggestion
# Infinitely smooth "bump" functions
```

**Review by fpvandoorn — APPROVED** [review]
LGTM

File: Mathlib/Analysis/Calculus/BumpFunctionInner.lean
Comment (urkud) [reply Mathlib/Analysis/Calculus/BumpFunctionInner.lean:1]: All parts of this file are in the new subfolder.

**Comment by sgouezel** [comment]
This looks good to me. I am not even sure we need to wait for port-complete, as lemmas are not deleted and are still properly aligned.
bors d+

File: Mathlib/Analysis/Calculus/BumpFunctionInner.lean
Comment (fpvandoorn) [reply Mathlib/Analysis/Calculus/BumpFunctionInner.lean:1]: Oops, I missed that this file was deleted, my bad.

**Comment by urkud** [comment]
bors merge

**Pull request closed without merging**